src/test/mir-opt/const_prop/slice_len.main.ConstProp.64bit.diff

```diff
@@ -27,14 +28,16 @@
                                            // + literal: Const { ty: &[u32; 3], val: Unevaluated(Unevaluated { def: WithOptConstParam { did: DefId(0:3 ~ slice_len[317d]::main), const_param_did: None }, substs: [], promoted: Some(promoted[0]) }) }
           _4 = _9;                         // scope 0 at $DIR/slice_len.rs:5:6: 5:19
           _3 = _4;                         // scope 0 at $DIR/slice_len.rs:5:6: 5:19
+          StorageLive(_10);                // scope 0 at $DIR/slice_len.rs:5:6: 5:19
+          _10 = _3;                        // scope 0 at $DIR/slice_len.rs:5:6: 5:19
           _2 = move _3 as &[u32] (Pointer(Unsize)); // scope 0 at $DIR/slice_len.rs:5:6: 5:19
           StorageDead(_3);                 // scope 0 at $DIR/slice_len.rs:5:18: 5:19
           StorageLive(_6);                 // scope 0 at $DIR/slice_len.rs:5:31: 5:32
           _6 = const 1_usize;              // scope 0 at $DIR/slice_len.rs:5:31: 5:32
--         _7 = Len((*_2));                 // scope 0 at $DIR/slice_len.rs:5:5: 5:33
+          _7 = const 3_usize;              // scope 0 at $DIR/slice_len.rs:5:5: 5:33
+          StorageDead(_10);                // scope 0 at $DIR/slice_len.rs:5:5: 5:33
 -         _8 = Lt(_6, _7);                 // scope 0 at $DIR/slice_len.rs:5:5: 5:33
 -         assert(move _8, "index out of bounds: the length is {} but the index is {}", move _7, _6) -> bb1; // scope 0 at $DIR/slice_len.rs:5:5: 5:33
-+         _7 = const 3_usize;              // scope 0 at $DIR/slice_len.rs:5:5: 5:33
 +         _8 = const true;                 // scope 0 at $DIR/slice_len.rs:5:5: 5:33
 +         assert(const true, "index out of bounds: the length is {} but the index is {}", const 3_usize, const 1_usize) -> bb1; // scope 0 at $DIR/slice_len.rs:5:5: 5:33
```

Comment:
Hmm, shouldn't be this assert removed, as it const true?

Reply:
It may be removed, but the test doesn't output the results of other passes, just the diff due to application of `ConstProp` one

Reply:
We run a `SimplifyBranches` pass right after `ConstProp` which will turn the assert into `goto -> bb1`